flink-java/src/main/java/org/apache/flink/api/java/operators/DataSink.java

```diff
@@ -208,6 +214,28 @@ public DataSink(DataSet<T> data, OutputFormat<T> format, TypeInformation<T> type
 		return this;
 	}
 
+	private void isValidSortKeyType(int field) {
```

Comment:
There are repeating yet similar code for isValidSortKeyType methods to check if sortable. Could the code be moved to utility class to be reused/ shared?